### PR TITLE
Update macbert_demo.py

### PR DESCRIPTION
**Pull request by KnightLancelot**
主要修复了“corrected_text = _text[:len(text)]”这里的问题——一旦一次传入一批数据进行纠错，同时句子中出现“[' ', '“', '”', '‘', '’', '\n', '…', '—', '擤']”中的字符，就会出现纠错后句子（长度）错误的问题，如：
传入句子["我不唉“看 琅擤琊榜” ", "你找到你最喜欢的工作，我也很高心。"]，返回“[('我不想“看 琅擤琊榜”琅琅琅看', [('唉', '想', 2, 3)]), ('你找到你最喜欢的工作，我也很高兴。', [('心', '兴', 15, 16)])]”。
所以这里通过“(torch.argmax(ids, dim=-1) * text_tokens.attention_mask[i])”修复这个问题。

同时还修复了“if i >= len(corrected_text):”“continue”的问题。其实我仔细测试过，应该是不存在“i >= len(corrected_text)”这种可能性的。

### CODE REVIEW

**Review by shibing624 — COMMENTED** [review]
ok
